components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdpManager.java

```diff
@@ -71,6 +71,17 @@ void updateResidentIdP(IdentityProvider identityProvider, String tenantDomain) t
      */
     List<IdentityProvider> getIdPs(String tenantDomain) throws IdentityProviderManagementException;
 
+     /**
+     * Retrieves registered Identity providers for a given tenant by Identity Provider name
+     *
+     * @param tenantDomain Tenant domain whose IdP names are requested
+     * @return Set of <code>IdentityProvider</code>. IdP names, primary IdP and home realm
+     * identifiers of each IdP
+     * @throws IdentityProviderManagementException Error when getting list of Identity Providers
+     */
+    List<IdentityProvider> getIdPsSearch(String tenantDomain, String filter)
```

Comment:
Hi @senthalan 

I have already added the implementation for the above method in "IdentityProviderManager.java" file. Please confim and elaborate if my understanding is correct or not.

Reply:
yes, you are I have implemented the newly added method in the IdentityProviderManager which wso2 product ship by default. But other developers can register a customized IdpManager. So if they don't have this method implemented. It will cause a runtime error. So  we always have a default implementation for the newly added public interface methods

Reply:
make newly added method as default method.

Reply:
Hi @senthalan 

I have implemented the changes suggested by you. Request you to verify my changes.